### PR TITLE
Support partial capture

### PR DESCRIPTION
**Pull request by AnasNaouchi**
Adds support for partial capture into this library: https://docs.opn.ooo/capture

### CODE REVIEW

File: Omise.Examples/Examples/Charges.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public async Task Create__Create_With_Customer()
         public async Task Capture__Capture()
         {
             var charge = RetrieveUncapturedCharge();
-            charge = await Client.Charges.Capture(charge.Id);
```
Comment (danfowler) [inline Omise.Examples/Examples/Charges.cs]: @AnasNaouchi can we keep the original capture example while also adding the new capture-with-amount example?

Comment (AnasNaouchi) [reply Omise.Examples/Examples/Charges.cs]: Yes we can

File: Omise/Models/ChargeRequest.cs [hunk 2]
```diff
@@ -2,13 +2,15 @@
...
-    public class CreateChargeRequest : Request
+namespace Omise.Models
+{
+    public class CreateChargeRequest : Request
     {
         public string Customer { get; set; }
         public string Card { get; set; }
         public long Amount { get; set; }
+        [JsonProperty("authorization_type")]
+        public string AuthorizationType { get; set; }
```
Comment (danfowler) [inline Omise/Models/ChargeRequest.cs]: https://stackoverflow.com/questions/32853482/how-can-i-make-my-string-property-nullable

File: Omise/Models/ChargeRequest.cs [hunk 3]
```diff
@@ -23,15 +25,21 @@ public class CreateChargeRequest : Request
...
+    public class UpdateChargeRequest : Request
     {
         public string Description { get; set; }
         public IDictionary<string, object> Metadata { get; set; }
-    }
+    }
+    public class CaptureChargeRequest : Request
+    {
+        [JsonProperty("capture_amount")]
+        public int CaptureAmount { get; set; }
```
Comment (nuxzero) [inline Omise/Models/ChargeRequest.cs]: I think the amount should has `long` type.

File: Omise/Models/AuthType.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+using System.Runtime.Serialization;
+
+namespace Omise.Models
+{
+    public enum AuthType
+    {
+        [EnumMember(Value = null)]
+         None,
```
Comment (nuxzero) [inline Omise/Models/AuthType.cs]: Does it need `None` enum member to handle `null`? Can it handle from property declaration?

Comment (AnasNaouchi) [reply Omise/Models/AuthType.cs]: I followed the `FlowTypes.cs` enum structure

Comment (nuxzero) [reply Omise/Models/AuthType.cs]: Okay, can you change the name to plural name?

Comment (AnasNaouchi) [reply Omise/Models/AuthType.cs]: changed @nuxzero

**Review by nuxzero — APPROVED** [review]

**Review by danfowler — APPROVED** [review]

**Pull request merged**